lib/src/authentication/credential.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// ignore_for_file: import_of_legacy_library_into_null_safe
+
+import '../exceptions.dart';
+import '../source/hosted.dart';
+
+/// Token is a structure for storing authentication credentials for third-party
+/// pub registries. A token holds registry [url], credential [kind] and [token]
+/// itself.
+///
+/// Token could be serialized into and from JSON format structured like
+/// this:
+///
+/// ```json
+/// {
+///   "url": "https://example.com/",
+///   "token": "gjrjo7Tm2F0u64cTsECDq4jBNZYhco"
+/// }
+/// ```
+class Credential {
+  Credential({required this.url, required this.token, this.unknownFields});
+
+  /// Create [Credential] instance for bearer tokens.
+  Credential.bearer(this.url, this.token) : unknownFields = null;
```

Comment:
Let's call this `token`, maybe in the future we have one called `tokenFromEnvironmentVariable` or something like that :D
```suggestion
  Credential.token(this.url, this.token) : unknownFields = null;
```